backend/socket.io/listeners.js

```diff
@@ -10,10 +10,18 @@ const connectionListener = (socket, io) => {
       console.error('Invalid channelID - ', channelID);
       return;
     }
+    const users = clients.getClientsByChannel(channelID);
+    const userIDs = users && Object.keys(users);
 
-    clients.setClientToChannel(userID, channelID, socket.id);
-    socket.channelID = channelID;
-    socket.userID = userID;
+    if (!users || userIDs.length < 2) {
```

Comment:
I would prefer simple conditions and early exit for better readability.
for example:
```
if(userCount === 2) {
      const receiverSocket = io.sockets.sockets[socket.id];
      receiverSocket.emit('unauthorised-user');
      return;
}
// continue as usual
```

Reply:
instead of `unauthorised-user`, may be something like `limit-reached`